src/wp-includes/blocks.php

```diff
@@ -178,10 +178,70 @@ function register_block_style_handle( $metadata, $field_name ) {
 	return $result ? $style_handle : false;
 }
 
+/**
+ * Gets i18n schema for block's metadata read from `block.json` file.
+ *
+ * @since 5.9.0
+ *
+ * @return array The schema for block's metadata.
+ */
+function get_i18n_block_metadata_schema() {
+	static $i18n_block_schema;
+
+	if ( ! isset( $i18n_block_schema ) ) {
+		$i18n_block_schema = wp_json_file_decode( __DIR__ . '/block-i18n.json' );
+	}
+
+	return $i18n_block_schema;
+}
+
+/**
+ * Translates the provided setting value using its i18n schema.
+ *
+ * @since 5.9.0
+ *
+ * @param string|string[]|array[] $i18n_schema   I18n schema for the setting.
+ * @param string|string[]|array[] $setting_value Value for the setting.
+ * @param string                  $textdomain    Textdomain to use with translations.
+ *
+ * @return string|string[]|array[] Translated setting.
+ */
+function translate_setting_using_i18n_schema( $i18n_schema, $setting_value, $textdomain ) {
```

Comment:
Can't this logic be made reusable by `WP_Theme_JSON_Resolver`?

I assumed that any new function introduced here could also be used in `extract_paths_to_translate()`/`get_fields_to_translate`

Reply:
There is one feature missing for using it with `WP_Theme_JSON_Resolver`. It's the special handling for `*` like:

<img width="938" alt="Screen Shot 2021-07-23 at 11 43 43" src="https://user-images.githubusercontent.com/699132/126764910-9576a32f-6af9-450a-8984-7b5aa1a8a4bc.png">

In theory, it should be possible to get rid of `extract_paths_to_translate`, `get_fields_to_translate` and `translate_theme_json_chunk`.

Reply:
I refactored code to use `translate_settings_using_i18n_schema` with both `block.json` and `theme.json`.